lib/widgets/home_header_search.dart

```diff
@@ -69,51 +69,44 @@ class _HomeHeaderSearchState extends State<HomeHeaderSearch> {
           fontSize: 16.0),
       height: 40.0,
     );
-    return AppBar(
-      elevation: 0.0,
-      titleSpacing: 0.0,
-      backgroundColor: widget.color,
-      actions: widget.actions,
-      title: Container(
-        margin: widget.margin,
-        child: Stack(
-          children: <Widget>[
-            textInput,
-            Positioned(
+    return Stack(children: [
+      Container(margin: EdgeInsets.only(left: 20, right: 50), child: textInput),
+      Positioned(
+        child: Container(
+          child: Icon(
+            Icons.search,
+            color: Colors.white,
+          ),
+          width: 16.0,
+          height: 16.0,
+        ),
+        left: 32.0,
+        top: 12.0,
+      ),
+      _showClose == true
+          ? Positioned(
               child: Container(
-                child: Icon(
-                  Icons.search,
-                  color: Colors.white,
+                child: RawMaterialButton(
+                  onPressed: () {
+                    setState(() {
+                      widget.textEditingController!.clear();
+                      _showClose = false;
+                    });
+                  },
+                  shape: CircleBorder(),
+                  child: Icon(Icons.close, color: Colors.white),
+                  padding: EdgeInsets.all(0.0),
                 ),
-                width: 16.0,
-                height: 16.0,
+                width: 50.0,
+              ),
+              right: 0.0,
+            )
+          : Positioned(
+              child: Row(
+                children: widget.actions!,
               ),
-              left: 12.0,
-              top: 12.0,
+              right: 0.0,
             ),
-            _showClose == true
-                ? Positioned(
-                    child: Container(
-                      child: RawMaterialButton(
-                        onPressed: () {
-                          setState(() {
-                            widget.textEditingController!.clear();
-                            _showClose = false;
-                          });
-                        },
-                        shape: CircleBorder(),
-                        child: Icon(Icons.close, color: Colors.white),
-                        padding: EdgeInsets.all(0.0),
-                      ),
-                      width: 40.0,
-                      height: 40.0,
-                    ),
-                    right: 8.0,
-                  )
-                : Container(height: 40.0),
-          ],
-        ),
-      ),
-    );
+    ]);
```

Comment:
I think this will be much cleaner if we remove `_showClose` and show close UI with `onClose` function. Would be great if you can refactor this.